tests/rptest/tests/rpk_cluster_test.py

```diff
@@ -184,3 +187,61 @@ def test_cluster_down(self):
             pass
         else:
             assert False, f"Unexpected success: '{r}'"
+
+    @cluster(num_nodes=3)
+    def test_upload_and_query_cluster_license_rpk(self):
+        """
+        Test uploading and retrieval of license via rpk
+        """
+        license = os.environ.get("REDPANDA_SAMPLE_LICENSE", None)
+        if license is None:
+            is_ci = os.environ.get("CI", "false")
+            assert is_ci == "false"
+            self.logger.info(
+                "Skipping test, REDPANDA_SAMPLE_LICENSE env var not found")
+            return
+
+        with tempfile.NamedTemporaryFile() as tf:
+            with open(tf.name, 'w') as f:
+                f.write(license)
+                f.seek(0)
+
+                output = self._rpk.license_set(tf.name)
+                assert "Successfully uploaded license" in output
+
+        def condition():
+            output = self._rpk.license_info()
+
+            expected = '''LICENSE INFORMATION:
+====================
+Organization:    redpanda-testing
+Type:            enterprise'''
+            if expected in output:
+                return True
+
+            return False
+
+        wait_until(condition,
+                   timeout_sec=10,
+                   backoff_sec=1,
+                   retry_on_exc=True,
+                   err_msg="unable to retrieve license information")
+
+    @cluster(num_nodes=3)
+    def test_upload_cluster_license_error(self):
+        with expect_exception(RpkException,
+                              lambda e: "Internal Server Error" in str(e)):
+            license = os.environ.get("REDPANDA_SAMPLE_LICENSE", None)
```

Comment:
This license env check stuff could be bundled in a util check

Reply:
Done, let me know what do you think.